gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALRasterSource.scala

```diff
@@ -137,3 +121,11 @@ case class GDALRasterSource(
     readBounds(bounds, 0 until bandCount)
   }
 }
+
+object GDALRasterSource {
+  def apply(path: String): GDALRasterSource =
+    GDALRasterSource(GDALDataPath(path))
+
+  def apply(path: String, options: GDALWarpOptions): GDALRasterSource =
+    GDALRasterSource(GDALDataPath(path), options)
```

Comment:
We've gone heavy on `apply` overloads in the past and it produced a lot of indirection. I wonder if having implicit `String => GDALDataPath` in the companion object of `GDALDataPath` would be better? It would at least cover the interactive case and would replace a lot of `apply` methods.

@moradology @pomadchin

Reply:
After thinking about it, I feel that these types of `apply`s are okay. They don't perform any real logic, and they return a `GDALRasterSource` in their body instead of calling another `apply`. I think `apply` methods like these are fine since they're essentially the same as an implicit method - all they do is provide a some convenience. It's when there's actual work being done in the `apply`s and they start calling each other that problems start emerging.

With the above being said, I wouldn't be opposed to using implicits, but I also don't think there's anything wrong with the current code.

Reply:
@echeipesh I would like to try a new approach, we can always revert back to overloads. The approach with `String => Path` implicit with have in `HadoopGeoTiffRDD.read{Singleband|Multiband}` overloads, the case is that we already have too many overloads and we wanted everything to work not only with `Path`  but also with `String`: https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/store/hadoop/HadoopGeoTiffRDD.scala#L116 and implicit method is for some reason here: https://github.com/locationtech/geotrellis/blob/master/store/src/main/scala/geotrellis/store/hadoop/package.scala#L24

Even though it is not in a companion object - it never created problems (since it is a pretty straightforward implicit) and it made the code much more transparent and easier to read (as it reduced overloads number two times).